### ISSUE TITLE
Put on crates.io, make the API stable enough to use as a standalone library

### ISSUE THREAD

**Issue by nukep** [nukep/llamadb#19]
I don't feel that LlamaDB is at a point where it can serve as a standalone library yet. The library crate is currently a bunch of hacks that expose internals such as parser modules.

This issue needs to wait on #4 so that developers have a somewhat reasonable API to work with, as well as a file-based DB. tempdb won't be around for long!

Until then, the `llamadb` crate has been reserved on crates.io.

---

Nice-to-haves for the library:
- A `sql!()` macro that statically validates and compiles the SQL query (to an extent).
  - This may be hard to do without schema information.
  - At least have it validate the syntax, possibly compile into AST?
- A C API, for use by other programming languages.
